hcloud/load_balancers.go

```diff
@@ -97,13 +121,19 @@ func (l *loadBalancers) EnsureLoadBalancer(
 	metrics.OperationCalled.WithLabelValues(op).Inc()
 
 	var (
-		reload bool
-		lb     *hcloud.LoadBalancer
-		err    error
+		reload        bool
+		lb            *hcloud.LoadBalancer
+		err           error
+		selectedNodes []*corev1.Node
 	)
 
-	nodeNames := make([]string, len(nodes))
-	for i, n := range nodes {
+	selectedNodes, err = matchNodeSelector(svc, nodes)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %v", op, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("%s: %w", op, err)
```

If you wrap errors, consumers can actually unwrap & inspect the the error later on if thats what they want to do.

https://go.dev/blog/go1.13-errors